[SERVER UI] FTP Parsing Workaround

nuno-chicoria opened the pull request:
## What changes are proposed in this pull request?

The ftp.nlst() function at line 106 creates a list that starts with "." and "..". These two items are not objects so, when accessing through FTP, MLflow throws an error about not being able to access those files. In this workaround, we iterate the list starting at index 2.

Fixes #1338 

## How is this patch tested?

TBD

## Release Notes

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

Fixes bug where artifacts could not be visualised when storing them through an FTP connection on a remote server.

### What component(s) does this PR affect?

- [X] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [X] Tracking
- [ ] Projects
- [X] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

nuno-chicoria committed:
FTP Parsing Workaround

The ftp.nlst() function at line 106 creates a list that starts with "." and "..". These two items are not objects so, when accessing through FTP, MLflow throws an error about not being able to access those files. In this workaround, we iterate the list starting at index 2.

mparkhe commented on mlflow/store/artifact/ftp_artifact_repo.py:
```diff
@@ -105,7 +105,7 @@ def list_artifacts(self, path=None):
             artifact_files = ftp.nlst(list_dir)
             infos = []
-            for file_name in artifact_files:
+            for file_name in artifact_files[2:]:
```
Starting from element # 2 sounds dangerous. Can we instead skip if `file_name` is `"."` or `".."`

nuno-chicoria replied:
Thank you for the input. I removed the `[2:]` and added a `list(filter(lambda x: x != "." and x != "..", artifact_list))` before to remove any `"."` and `".."`.

nuno-chicoria replied:
Hi @mparkhe! Could you give us a follow up on this? Do you think it is relevant to change and merge with MLflow master? Is there anything we can do to help more with the process?

nuno-chicoria committed:
Change FTP Workaround Method

Instead of parsing the list from index 2 the list is filtered beforehand to remove "." and "..".

nuno-chicoria committed:
Wrong Variable Name

Commit:
Merge remote-tracking branch 'origin/master' into nuno-chicoria-master

mparkhe commented:
fixes #1814 #1338
